### PR TITLE
refactor: manage CourseCreator operations through a new manager

### PR DESCRIPTION
**Pull request by shadinaif**
refactor: manage CourseCreator operations through a new manager

### CODE REVIEW

**Review by OmarIthawi — COMMENTED** [review]
Thanks Shadi! Some minor comments.

Overall, looks good!

File: futurex_openedx_extensions/helpers/course_creator_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            is_all_orgs = self.is_all_orgs()
+        except FXCodedException as exc:
+            if exc.code == FXExceptionCodes.COURSE_CREATOR_NOT_FOUND.value:
+                return
+            raise exc
+
+        if is_all_orgs:
+            return
+
+        orgs = [org.lower() for org in orgs]
```
[the remark this answers is not in the document]
Comment (shadinaif) [reply futurex_openedx_extensions/helpers/course_creator_wrapper.py:186]: that's right, but `edx-platform` works on MySQL's default collation `ci` (case-insensitive). So, SQL `IN` is case-insensitive for sure. No need to add the overhead of multiple `ILIKE` in the query. Is there any chance to change the collation in the future? I highly doubt it

File: futurex_openedx_extensions/helpers/course_creator_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        organizations_many_to_many = CourseCreator.organizations.through
+        organizations_many_to_many.objects.filter(
+            coursecreator_id=self.db_record.id,
+            organization__short_name__in=orgs,
```
[the remark this answers is not in the document]
Comment (shadinaif) [reply futurex_openedx_extensions/helpers/course_creator_wrapper.py:190]: please see my comment [here](https://github.com/nelc/futurex-openedx-extensions/pull/71#discussion_r1762430680)

File: futurex_openedx_extensions/helpers/course_creator_wrapper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (OmarIthawi) [reply futurex_openedx_extensions/helpers/course_creator_wrapper.py:186]: great, in thise case this line is not needed anyway.

File: futurex_openedx_extensions/helpers/course_creator_wrapper.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (OmarIthawi) [reply futurex_openedx_extensions/helpers/course_creator_wrapper.py:190]: sure sure, no need for the complication then. thanks Shadi.

File: futurex_openedx_extensions/helpers/course_creator_wrapper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shadinaif) [reply futurex_openedx_extensions/helpers/course_creator_wrapper.py:186]: oh that's true, it's not needed at all! thank you @OmarIthawi

**Review by OmarIthawi — APPROVED** [review]
Great! Thanks @shadinaif!!

**Pull request merged**